packages/itwin/reports-config-widget/src/widget/components/ExtractionStatus.tsx

```diff
@@ -0,0 +1,70 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import * as React from "react";
+import { useEffect, useState } from "react";
+import "./ExtractionStatus.scss";
+import { STATUS_CHECK_INTERVAL } from "./Constants";
+import { StartingExtractionState } from "./ExtractionStates/StartingExtractionState";
+import { FetchingUpdateExtractionState } from "./ExtractionStates/FetchingUpdateExtractionState";
+import { RunningExtractionState } from "./ExtractionStates/RunningExtractionState";
+import { FailedExtractionState } from "./ExtractionStates/FailedExtractionState";
+import { QueuedExtractionState } from "./ExtractionStates/QueuedExtractionState";
+import { SucceededExtractionState } from "./ExtractionStates/SucceededExtractionState";
+
+export enum ExtractionStates {
+  None,
+  Starting,
+  FetchingUpdate,
+  Queued,
+  Running,
+  Succeeded,
+  Failed,
+}
+
+interface ExtractionStatusProps {
+  state: ExtractionStates;
+  clearExtractionState: () => void;
+  children?: React.ReactNode;
+}
+
+export const ExtractionStatus = ({
+  state,
+  children,
+  clearExtractionState,
+}: ExtractionStatusProps) => {
+  const [fadeOut, setFadeOut] = useState<boolean>(false);
+
+  const onAnimationEnd = () => {
+    clearExtractionState();
+    setFadeOut(false);
+  };
+
+  useEffect(() => {
+    let timer: number;
+    switch (state) {
+      case ExtractionStates.Succeeded:
+      case ExtractionStates.Failed:
+        timer = window.setTimeout(() => setFadeOut(true), STATUS_CHECK_INTERVAL);
+    }
+    return () => window.clearTimeout(timer);
+  }, [state]);
+
+  switch (state) {
+    case ExtractionStates.Starting:
+      return StartingExtractionState();
```

Comment:
These are react components - switch them to jsx syntax.
```
<StartingExtractionState />
<SucceededExtractionState fadeOut={fadeOut} />
... etc
```

Reply:
Components syntax switched to jsx